PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -176,17 +238,24 @@ public function read($n)
      */
     public function write($data)
     {
+        $written = 0;
         $len = mb_strlen($data, 'ASCII');
-        while (true) {
-            if (is_null($this->sock)) {
+        
+        while ($written < $len) {
+
+            if (!is_resource($this->sock)) {
                 throw new AMQPRuntimeException('Broken pipe or closed connection');
             }
 
-            if (false === ($written = @fwrite($this->sock, $data))) {
+            set_error_handler(array($this, 'error_handler'));
+            $buffer = fwrite($this->sock, $data);
```

Comment:
Should the fwrite() be wrapped in a try catch?
error_hander() can throw an exception and if it does, restore_error_handler() won't get called.